unittests/checktime_tests.cpp

```diff
@@ -150,6 +150,36 @@ BOOST_AUTO_TEST_CASE( checktime_interrupt_test) { try {
 
 } FC_LOG_AND_RETHROW() }
 
+BOOST_AUTO_TEST_CASE( checktime_speculative_max_trx_test ) { try {
+   savanna_tester t;
+   t.produce_block();
+   t.create_account( "pause"_n );
+   t.set_code( "pause"_n, test_contracts::test_api_wasm() );
+   t.produce_block();
+
+   BOOST_CHECK_EXCEPTION( push_trx( t, test_pause_action<WASM_TEST_ACTION("test_checktime", "checktime_failure")>{},
+                                    0, 25, 500, false, fc::raw::pack(10000000000000000000ULL), "pause"_n ),
+                          tx_cpu_usage_exceeded, fc_exception_message_contains("reached node configured max-transaction-time") );
+
+   // test case where max-transaction-time = -1, fc::microseconds::maximum()
+   // Verify restricted to 150ms (on-chain max_transaction_cpu_usage)
+   BOOST_CHECK_EXCEPTION( push_trx( t, test_pause_action<WASM_TEST_ACTION("test_checktime", "checktime_failure")>{},
+                                     100000, UINT32_MAX, 10000, false, fc::raw::pack(10000000000000000000ULL), "pause"_n ),
```

Comment:
In the comment, maybe mentions `UINT32_MAX` is what  `max-transaction-time = -1` is converted to.